examples/1-getting-started.ipynb

```diff
@@ -0,0 +1,322 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Getting started with steps\n",
+    "\n",
+    "This notebook shows how to **create** steps, **fit** them to data, **transform** new data and take advantage of persistence"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load_ext autoreload\n",
+    "%autoreload 2\n",
+    "\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline\n",
+    "\n",
+    "from steps.base import Step, BaseTransformer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Grabbing some data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We'll import a dataset from scikit-learn for our experiments and divide it into training and test sets"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.datasets import load_digits\n",
+    "\n",
+    "digits = load_digits()\n",
+    "X_digits, y_digits = digits.data, digits.target\n",
+    "\n",
+    "num_train = int(0.8 * len(y_digits))\n",
+    "\n",
+    "sample_ids = np.random.permutation(len(y_digits))\n",
+    "\n",
+    "train_ids = sample_ids[:num_train]\n",
+    "test_ids = sample_ids[num_train:]\n",
+    "\n",
+    "print('{} samples for training'.format(len(train_ids)))\n",
+    "print('{} samples for test'.format(len(test_ids)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Steps communicate data between each other with plain **Python dictionaries**. This makes it easy to pass collections of **arbitrary data types** (Numpy arrays, Pandas dataframes, etc.). The basic structure is as follows (you can get much more fancy but we leave that to the next example)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_train = {'input':\n",
+    "                {\n",
+    "                     'X': X_digits[train_ids, :],\n",
+    "                     'y': y_digits[train_ids],\n",
+    "                }\n",
+    "            }\n",
+    "\n",
+    "data_test = {'input':\n",
+    "                {\n",
+    "                     'X': X_digits[test_ids, :],\n",
+    "                     'y': y_digits[test_ids],\n",
+    "                }\n",
+    "            }"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating steps\n",
+    "Let's create a simple step - first we define an **adapter** to tell it how to interpret its input data dictionary (this allows you to do lots of clever things but we'll stick to basics for now)"
```

Comment:
@mromaniukcdl as I understand you introduce `adapter` because running this without the `adapter` doesn't work. If so we should fix/implement simple option right away to be able make shift without this new concept in the very first tutorial.
Can you confirm that running without the `adapter` doesn't work?